pkg/disttask/framework/storage/table_test.go

```diff
@@ -93,12 +84,10 @@ func TestTaskTable(t *testing.T) {
 	require.Equal(t, task, task4[0])
 	require.GreaterOrEqual(t, task4[0].StateUpdateTime, task.StateUpdateTime)
 
-	prevState := task.State
-	task.State = proto.TaskStateRunning
-	retryable, err := gm.UpdateTaskAndAddSubTasks(ctx, task, nil, prevState)
+	err = gm.SwitchTaskStep(ctx, task, proto.TaskStateRunning, proto.StepOne, nil)
 	require.NoError(t, err)
-	require.True(t, retryable)
 
+	task.State = proto.TaskStateRunning
```

Comment:
how about updating state and step in `SwitchTaskStep`, and returning the new task?

Reply:
that's side effects we should avoid